packages/ubuntu_desktop_installer/test/installation_type_page_test.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+import 'package:ubuntu_desktop_installer/l10n.dart';
+import 'package:ubuntu_desktop_installer/pages/installation_type/installation_type_model.dart';
+import 'package:ubuntu_desktop_installer/pages/installation_type/installation_type_page.dart';
+import 'package:ubuntu_test/mocks.dart';
+import 'package:ubuntu_test/utils.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+
+import 'installation_type_page_test.mocks.dart';
+import 'widget_tester_extensions.dart';
+
+@GenerateMocks([InstallationTypeModel])
+void main() {
+  InstallationTypeModel buildModel({
+    InstallationType? installationType,
+    AdvancedFeature? advancedFeature,
+    bool? encryption,
+    String? productInfo,
+    String? existingOS,
+  }) {
+    final model = MockInstallationTypeModel();
+    when(model.installationType)
+        .thenReturn(installationType ?? InstallationType.erase);
+    when(model.advancedFeature)
+        .thenReturn(advancedFeature ?? AdvancedFeature.none);
+    when(model.encryption).thenReturn(encryption ?? false);
+    when(model.productInfo).thenReturn(productInfo ?? '');
+    when(model.existingOS).thenReturn(existingOS);
+    return model;
+  }
+
+  Widget buildPage(InstallationTypeModel model) {
+    return ChangeNotifierProvider<InstallationTypeModel>.value(
+      value: model,
+      child: InstallationTypePage(),
+    );
+  }
+
+  Widget buildApp(InstallationTypeModel model) {
+    return MaterialApp(
+      localizationsDelegates: localizationsDelegates,
+      home: Wizard(
+        routes: {'/': (_) => buildPage(model)},
+        onNext: (settings) => '/',
+      ),
+    );
+  }
+
+  testWidgets('reinstall', (tester) async {
+    final model = buildModel(
+      installationType: InstallationType.reinstall,
```

Comment:
I'm not sure I understand why the model's `installationType` is initialized to `InstallationType.reinstall`, if we're checking the behaviour of selecting the option from the UI (same question for the next 3 tests, `alongside`, `erase` and `manual`)?

Reply:
i removed these lines because the initial type doesn't actually matter. this is testing that the correct installation type is passed to the model when the respective radio button is clicked